mono/metadata/reflection.c

```diff
@@ -194,11 +194,16 @@ static MonoMethod * inflate_method (MonoReflectionType *type, MonoObject *obj);
 static guint32 create_typespec (MonoDynamicImage *assembly, MonoType *type);
 static void init_type_builder_generics (MonoObject *type);
 
-#define RESOLVE_TYPE(type) do { type = (MonoObject *)mono_reflection_type_resolve_user_types ((MonoReflectionType*)type); } while (0)
-#define RESOLVE_ARRAY_TYPE_ELEMENT(array, index) do {	\
-	MonoReflectionType *__type = mono_array_get (array, MonoReflectionType*, index);	\
-	__type = mono_reflection_type_resolve_user_types (__type);	\
-	mono_array_set (arr, MonoReflectionType*, index, __type);	\
+#define RESOLVE_TYPE(type) do {						\
+	MonoError __error;						\
+	type = (MonoObject *)mono_reflection_type_resolve_user_types ((MonoReflectionType*)type, &__error); \
+	mono_error_raise_exception (&__error); /* FIXME don't raise here */ \
+} while (0)
+#define RESOLVE_ARRAY_TYPE_ELEMENT(array, index, error) do {		\
+	MonoReflectionType *__type = mono_array_get (array, MonoReflectionType*, index); \
+	__type = mono_reflection_type_resolve_user_types (__type, error); \
+	if (mono_error_ok (error))					\
```

Comment:
This check doesn't change behavior since it's storing on a fresh array full of zeros.